### PR TITLE
Confirmation for navigating away from edit event/series

### PR DESCRIPTION
**Pull request by k3ntako**

### CODE REVIEW

File: client/pages/EventForm/EventForm.js [hunk 1]
```diff
@@ -97,11 +122,13 @@ class EventForm extends Component {
 
     if(this.props.match.path.includes("/edit")){
       this.state.event.update(name, datetime, location, draftNote, assignments).then( (event) => {
+        window.removeEventListener('beforeunload', this.beforeunload);
```
Comment (mienaikoe) [inline client/pages/EventForm/EventForm.js]: you can just let componentWillUnmount handle this

File: client/pages/EventForm/EventForm.js [hunk 2]
```diff
@@ -97,11 +122,13 @@ class EventForm extends Component {
...
         this.props.history.push(`/events/${this.props.match.params.id}`);
       });
     }else {
       const assignmentNames = assignments.map( assignment => assignment.name );
       Event.create(name, datetime, location, draftNote, assignmentNames).then( (event) => {
+        window.removeEventListener('beforeunload', this.beforeunload);
```
Comment (mienaikoe) [inline client/pages/EventForm/EventForm.js]: same as above

File: client/pages/SeriesForm/SeriesForm.js [hunk 3]
```diff
@@ -181,6 +209,7 @@ class SeriesForm extends Component {
                 key="draft"
                 label="Notes for Event"
                 onChange={this.eventNoteChangeHandler}
+                setHasChanged={this.setHasChanged.bind(this)}
```
Comment (mienaikoe) [inline client/pages/SeriesForm/SeriesForm.js]: do the binding in the constructor so you don't have to call bind a bunch of times at render

File: client/components/EventTextEditor/index.js [hunk 4]
```diff
@@ -4,8 +4,16 @@ import styles from './EventEditor.css';
 
 
 export default (props) => {
+  const onChange = (editorState) => {
+    if(editorState.getSelection().getHasFocus()){
+      props.setHasChanged();
+      return props.onChange(editorState)
```
Comment (mienaikoe) [inline client/components/EventTextEditor/index.js]: can't props.onChange call setHasChanged on its own? I don't think you need to call them both back-to-back

**Comment by mienaikoe** [comment]
Also I suggest making the target of this PR dev instead of 1.1.X

We're not demoing this today.

File: client/components/EventTextEditor/index.js (hunk 4 above)
Comment (k3ntako) [reply client/components/EventTextEditor/index.js]: I moved `setHasChanged()` up one level. is that what you meant?

Comment (mienaikoe) [reply client/components/EventTextEditor/index.js]: depends on what you mean by level

Comment (mienaikoe) [reply client/components/EventTextEditor/index.js]: ah yep. that's what I meant

File: client/pages/EventForm/EventForm.js [hunk 5]
```diff
@@ -84,6 +84,7 @@ class EventForm extends Component {
   }
 
   eventNoteChangeHandler(note){
+    this.setHasChanged();
     this.setState({ note });
```
Comment (mienaikoe) [inline client/pages/EventForm/EventForm.js]: Doesn't this effectively do two `setState` calls? Collapse this to one. It looks like you did this for all the other handlers.

Comment (k3ntako) [reply client/pages/EventForm/EventForm.js]: "did this" meaning I did collapse it, right?  Meaning this is the only place to fix it?

**Pull request merged**